staging/src/k8s.io/apiextensions-apiserver/pkg/cmd/server/testing/testserver.go

```diff
@@ -124,7 +124,11 @@ func StartTestServer(t Logger, _ *TestServerInstanceOptions, customFlags []strin
 	fs := pflag.NewFlagSet("test", pflag.PanicOnError)
 
 	featureGate := utilfeature.DefaultMutableFeatureGate
+
+	// Configure the effective version.
 	effectiveVersion := utilversion.DefaultKubeEffectiveVersion()
+	effectiveVersion.SetEmulationVersion(featureGate.EmulationVersion())
+
```

Comment:
@Jefftree does this looks right? I still needed it after https://github.com/kubernetes/kubernetes/pull/127643 merged and I rebased to master.

Reply:
Ah this is for apiextension testserver which I missed in https://github.com/kubernetes/kubernetes/pull/127643. Is the CI passing at the moment? I think we need will need to change it to pass binaryversion and emulatedversion as testserver options like in https://github.com/kubernetes/kubernetes/pull/127643. I'll look a bit more into it.

Reply:
SG. Do you want to hold this for that fix?

Reply:
Wait actually looking at this again I like this approach more than https://github.com/kubernetes/kubernetes/pull/127643 since we don't need to explicitly pass in the parameters and can infer it from the feature gate settings. LGTM